git_machete/cmd.py

```diff
@@ -339,359 +339,1027 @@ def popen_git(cli_ctxt: CommandLineContext, git_cmd: str, *args: str, **kwargs:
 
 # Manipulation on definition file/tree of branches
 
-managed_branches: List[str] = []
-down_branches: Dict[str, List[str]] = {}  # TODO (#110): default dict with []
-up_branch: Dict[str, str] = {}  # TODO (#110): default dict with None
-indent: Optional[str] = None
-roots: List[str] = []
-annotations: Dict[str, str] = {}
-
+branch_defs_by_sha_in_reflog: Optional[Dict[str, Optional[List[Tuple[str, str]]]]] = None
 
-def expect_in_managed_branches(b: str) -> None:
-    if b not in managed_branches:
-        raise MacheteException(
-            f"Branch `{b}` not found in the tree of branch dependencies.\nUse `git machete add {b}` or `git machete edit`")
+BRANCH_DEF = Tuple[str, str]
+empty_line_status: Optional[bool] = None
 
 
-def expect_at_least_one_managed_branch() -> None:
-    if not roots:
-        raise_no_branches_error()
+class MacheteClient:
 
+    cli_ctxt: CommandLineContext = CommandLineContext()
```

Comment:
Hmmm this is now a static field... is it still needed?

Reply:
I'm not even sure how Python treats an override like that... whether it's possible to have two separate class&instance attributes with the same name :/

Reply:
Anyway, this should probably be an instance attribute

Reply:
cli_ctxt as class attribute is not used anywhere, hence it should be removed.